src/main/java/net/sf/jabref/logic/importer/fetcher/MrDLibFetcher.java

```diff
@@ -0,0 +1,145 @@
+/**
+ *
+ */
+package net.sf.jabref.logic.importer.fetcher;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+import javax.net.ssl.SSLContext;
+
+import net.sf.jabref.logic.importer.EntryBasedFetcher;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.ParserResult;
+import net.sf.jabref.logic.importer.fileformat.MrDLibImporter;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.BuildInfo;
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContexts;
+
+/**
+ *  This class is responible to get the recommendations from MDL
+ *
+ */
+public class MrDLibFetcher implements EntryBasedFetcher {
+
+    private static final String NAME = "MDL_FETCHER";
+    private static final Log LOGGER = LogFactory.getLog(MrDLibFetcher.class);
+    private final String LANGUAGE;
+    private final String VERSION;
+    private final JabRefPreferences prefs = JabRefPreferences.getInstance();
+
+    public MrDLibFetcher() {
+        LANGUAGE = prefs.get(JabRefPreferences.LANGUAGE);
```

Comment:
Please pass the language as a parameter to the constructor. We try to avoid preference access in the logic part.